ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/CounterNamingStrategy.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.ucte.converter;
+
+import com.google.auto.service.AutoService;
+import com.powsybl.iidm.network.*;
+import com.powsybl.ucte.converter.util.UcteConstants;
+import com.powsybl.ucte.converter.util.UcteConverterHelper;
+import com.powsybl.ucte.network.UcteCountryCode;
+import com.powsybl.ucte.network.UcteElementId;
+import com.powsybl.ucte.network.UcteNodeCode;
+import com.powsybl.ucte.network.UcteVoltageLevelCode;
+
+import java.util.*;
+
+/**
+ * @author Clément LECLERC {@literal <clement.leclerc@rte-france.com>}
+ */
+@AutoService(NamingStrategy.class)
+public class CounterNamingStrategy extends AbstractNamingStrategy {
+
+    private int voltageLevelCounter;
+
+    @Override
+    public String getName() {
+        return "Counter";
+    }
+
+    @Override
+    public void initialiseNetwork(Network network) {
+        voltageLevelCounter = 0;
+        network.getVoltageLevelStream()
+                .forEach(this::processVoltageLevel);
+
+        network.getBranchStream().forEach(this::generateUcteElementId);
+        network.getDanglingLineStream().forEach(this::generateUcteElementId);
+    }
+
+    private void processVoltageLevel(VoltageLevel voltageLevel) {
+        Iterator<Bus> buslist = voltageLevel.getBusBreakerView().getBuses().iterator();
+        for (int i = 0; buslist.hasNext(); i++) {
+            Bus bus = buslist.next();
+            char orderCode = UcteConverterHelper.getOrderCode(i);
+            generateUcteNodeId(bus.getId(), voltageLevel, orderCode);
+        }
+
+        voltageLevel.getBusBreakerView().getSwitches()
+                .forEach(this::generateUcteElementId);
+        voltageLevelCounter++;
+    }
+
+    private UcteNodeCode generateUcteNodeId(String busId, VoltageLevel voltageLevel, char orderCode) {
+        if (UcteNodeCode.isUcteNodeId(busId)) {
+            return changeOrderCode(busId, orderCode);
+        }
+        return createNewUcteNodeId(busId, voltageLevel, orderCode);
+    }
+
+    private UcteNodeCode changeOrderCode(String busId, char orderCode) {
+        UcteNodeCode newNodeCode = UcteNodeCode.parseUcteNodeCode(busId).orElseThrow();
+        newNodeCode.setBusbar(orderCode);
+        ucteNodeIds.put(busId, newNodeCode);
+        return newNodeCode;
+    }
+
+    private UcteNodeCode createNewUcteNodeId(String busId, VoltageLevel voltageLevel, char orderCode) {
+        String newNodeId = String.format("%05d", voltageLevelCounter);
```

Comment:
Maybe we can use [A-Z0-9] codes instead?
We could convert the number in base 36, but I think it would be better to have the letters first instead of the digits.
(=> AAAAA to AAAAZ, AAAA0 to AAAA9, then AAAABA...)

Reply:
(But it is not required.)

Reply:
With numbers is ok for now. Let's see that later if one day needed.